tests/phpunit/integration/Core/Tags/Google_Tag_Gateway/REST_Google_Tag_Gateway_ControllerTest.php

```diff
@@ -360,6 +369,88 @@ public function test_get_gtg_server_requirement_status__requires_authenticated_a
 		$this->assertEquals( 'rest_forbidden', $response->get_data()['code'] );
 	}
 
+	public function test_set_settings_isGTGDefault() {
+		remove_all_filters( 'googlesitekit_rest_routes' );
+		$this->controller->register();
+		$this->register_rest_routes();
+		// Set up the site and admin user to make a successful REST request.
+		$this->grant_manage_options_permission();
+
+		$this->settings->register();
+
+		// Test explicitly setting isGTGDefault to false.
+		$request = new WP_REST_Request( 'POST', '/' . REST_Routes::REST_ROOT . '/core/site/data/gtg-settings' );
+		$request->set_body_params(
+			array(
+				'data' => array(
+					'settings' => array(
+						'isEnabled'    => false,
+						'isGTGDefault' => false,
+					),
+				),
+			)
+		);
+
+		$response = rest_get_server()->dispatch( $request );
+		$this->assertEqualSetsWithIndex(
+			array(
+				'isEnabled'             => false,
+				'isGTGHealthy'          => null,
+				'isScriptAccessEnabled' => null,
+				'isGTGDefault'          => false,
+			),
+			$response->get_data()
+		);
+
+		// Test explicitly setting isGTGDefault to true.
+		$request = new WP_REST_Request( 'POST', '/' . REST_Routes::REST_ROOT . '/core/site/data/gtg-settings' );
+		$request->set_body_params(
+			array(
+				'data' => array(
+					'settings' => array(
+						'isEnabled'    => false,
+						'isGTGDefault' => true,
+					),
+				),
+			)
+		);
+
+		$response = rest_get_server()->dispatch( $request );
+		$this->assertEqualSetsWithIndex(
+			array(
+				'isEnabled'             => false,
+				'isGTGHealthy'          => null,
+				'isScriptAccessEnabled' => null,
+				'isGTGDefault'          => true,
+			),
+			$response->get_data()
+		);
+
+		// Test setting both isEnabled and isGTGDefault - REST API allows explicit control.
+		$request = new WP_REST_Request( 'POST', '/' . REST_Routes::REST_ROOT . '/core/site/data/gtg-settings' );
+		$request->set_body_params(
+			array(
+				'data' => array(
+					'settings' => array(
+						'isEnabled'    => true,
+						'isGTGDefault' => true,
+					),
+				),
+			)
+		);
+
+		$response = rest_get_server()->dispatch( $request );
+		$this->assertEqualSetsWithIndex(
+			array(
+				'isEnabled'             => true,
+				'isGTGHealthy'          => null,
+				'isScriptAccessEnabled' => null,
+				'isGTGDefault'          => true, // REST API allows explicit control.
+			),
+			$response->get_data()
+		);
```

Comment:
Let's use separate tests for these (or data provider) rather than putting all in one.

Reply:
Leaving this for @hussain-t to update.